engine/orchestration/src/com/cloud/agent/manager/AgentManagerImpl.java

```diff
@@ -986,33 +986,28 @@ public Answer easySend(final Long hostId, final Command cmd) {
     }
 
     @Override
-    public boolean reconnect(final long hostId) {
+    public void reconnect(final long hostId) throws CloudRuntimeException, AgentUnavailableException{
```

Comment:
@rafaelweingartner 
The reconnect method is a method in the interface AgentManager.java, This interface is also implemented by ClusteredAgentManagerImpl.java in which propagateAgentEvent method throws a AgentUnavailableException.  So i had to add this to the interface and also to the implementing class method.